x/launch/keeper/msg_server_settle_request.go

```diff
@@ -0,0 +1,92 @@
+package keeper
+
+import (
+	"context"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/tendermint/spn/x/launch/types"
+)
+
+func (k msgServer) SettleRequest(
+	goCtx context.Context,
+	msg *types.MsgSettleRequest,
+) (*types.MsgSettleRequestResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	chain, found := k.GetChain(ctx, msg.ChainID)
+	if !found {
+		return nil, sdkerrors.Wrap(types.ErrChainNotFound, msg.ChainID)
+	}
+
+	if chain.LaunchTriggered {
+		return nil, sdkerrors.Wrap(types.ErrTriggeredLaunch, msg.ChainID)
+	}
+
+	coordAddress := k.profileKeeper.GetCoordinatorAddressFromID(ctx, chain.CoordinatorID)
+	if msg.Coordinator != coordAddress {
+		return nil, sdkerrors.Wrap(types.ErrNoAddressPermission, msg.Coordinator)
+	}
+
+	// first check if all requests exist
+	requests := make([]types.Request, len(msg.RequestIDs))
+	for _, requestID := range msg.RequestIDs {
+		request, found := k.GetRequest(ctx, msg.ChainID, requestID)
+		if !found {
+			return nil, sdkerrors.Wrapf(types.ErrRequestNotFound,
+				"request %d for chain %s not found",
+				requestID,
+				msg.ChainID,
+			)
+		}
+		requests = append(requests, request)
+	}
+
+	// perform request action
+	for _, request := range requests {
+		k.RemoveRequest(ctx, msg.ChainID, request.RequestID)
+		if msg.Approve {
+			err := applyRequest(ctx, k.Keeper, msg, request)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return &types.MsgSettleRequestResponse{}, nil
+}
+
+// applyRequest approve the request and perform
+// the launch information changes
+func applyRequest(
+	ctx sdk.Context,
+	k Keeper,
+	msg *types.MsgSettleRequest,
+	request types.Request,
+) error {
+	cdc := codectypes.NewInterfaceRegistry()
+	types.RegisterInterfaces(cdc)
+
+	var content types.RequestContent
+	if err := cdc.UnpackAny(request.Content, &content); err != nil {
+		return sdkerrors.Wrap(types.ErrInvalidRequestContent, err.Error())
+	}
+
+	switch c := content.(type) {
+	case *types.GenesisAccount:
```

Comment:
Instead of using a switch, I think we can add a `Apply(context) error` method to the `RequestContent` interface